sources/predator_factory.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+Name: predator_factory.cpp
+Purpose: Abstract factory that houses genetics for predators. 
+Last edit: 11-11-2019
+Last editor: MG
+*/
+
+#include "predator_factory.h"
+
+predator_factory::predator_factory()
+{
+
+}
+
+predator_factory::~predator_factory()
+{
+
+}
+
+/*
+Name: punnett_square
+Purpose: everytime it is called, it returns the gene of one parent. If they're the same gene, it just returns the first.
+         If the genes are different, it has a 50/50 percent chance of which it returns. Follows Mendel's genetics.
+Trace: Traces to Epic 3, Acceptance Criteria 3
+Parameters: two characters of same genotype from one predator parent.
+Returns: one character of same genotype from one predator parent to pass on to the child.
+*/
+char predator_factory::punnett_square(char gene_one, char gene_two)
+{
+    if (gene_one != gene_two)
+    {
+        srand(time(NULL));
```

Comment:
It's a good idea to `srand(...)` one time, probably in `init_sim()`

Reply:
Agreed, the way it is now though, it isn't global to anything. So, I don't see too many problems with it. Other than here and in the seed.cpp, where else do we need a random?